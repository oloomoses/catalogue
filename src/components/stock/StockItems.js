/* eslint-disable react/forbid-prop-types */
import React from 'react';
import { Link } from 'react-router-dom';
import { Card } from 'react-bootstrap';
import PropTypes from 'prop-types';

const StockItems = ({ company }) => {
  const {
    changes,
    changesPercentage,
    companyName,
    price,
    ticker,
  } = company;
  return (
    <Card className="stockCard">
      <Card.Body>
        <Card.Title className="ticker">{ticker}</Card.Title>
        <Card.Subtitle className="companyName">{companyName}</Card.Subtitle>
        <Card.Text className="mt-3">
          <span className="profit">Price</span>
          {price}
          {' '}
          USD
        </Card.Text>
        <div>
          <span className="profit">Profit</span>
          {' '}
          <span className={changes > 0 ? 'text-success' : 'text-danger'}>
            <span className="changes">{changes}</span>
            {changesPercentage}
          </span>
        </div>
      </Card.Body>
      <Link to={`/company/${ticker}`} className="btn btn-primary">More... </Link>
    </Card>
  );
};

StockItems.propTypes = {
  company: PropTypes.object.isRequired,
};

export default StockItems;
