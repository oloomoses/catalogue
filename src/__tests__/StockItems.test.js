import renderer from 'react-test-renderer';
import StockItems from '../components/stock/StockItems';

describe('renders correctly', () => {
  it('matches the snapshot', () => {
    const company = {
      changes: 'BRK-A',
      changesPercentage: 2939,
      companyName: 'Berkshire Hathaway Inc',
      price: 408840,
      ticker: 'BRK-A',
    };
    const tree = renderer.create(<StockItems company={company} />);
    expect(tree).toMatchSnapshot();
  });
});
