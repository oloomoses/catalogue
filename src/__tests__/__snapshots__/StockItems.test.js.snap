// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly matches the snapshot 1`] = `
<div
  className="stockCard card"
>
  <div
    className="card-body"
  >
    <div
      className="ticker card-title h5"
    >
      BRK-A
    </div>
    <div
      className="companyName card-subtitle h6"
    >
      Berkshire Hathaway Inc
    </div>
    <p
      className="mt-3 card-text"
    >
      <span
        className="profit"
      >
        Price
      </span>
      408840
       
      USD
    </p>
    <div>
      <span
        className="profit"
      >
        Profit
      </span>
       
      <span
        className="text-danger"
      >
        <span
          className="changes"
        >
          BRK-A
        </span>
        2939
      </span>
    </div>
  </div>
  <a
    className="btn btn-primary"
    href="/company/BRK-A"
  >
    More... 
  </a>
</div>
`;
